wandb/sdk/artifacts/artifact.py

```diff
@@ -1300,7 +1300,8 @@ def add_reference(
                 automatic integrity validation. Disabling checksumming will speed up
                 artifact creation but reference directories will not iterated through so the
                 objects in the directory will not be saved to the artifact. We recommend
-                adding reference objects in the case checksumming is false.
+                setting `checksum=False` when adding reference objects, in which case
+                a new version will only be created if the reference URI changes.
```

Comment:
comment(non-blocking): for posterity -- just following up on the question I asked in the [previous PR](https://github.com/wandb/wandb/pull/7934#discussion_r1672685134)

> question: Checking, but in case you already did -- do we happen to have an existing test for this?

after digging a bit -- yep, looks like we do 😌 :
https://github.com/wandb/wandb/blob/d3329a6e201b2efebe6ce9e568b984fdedae1c98/tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts.py#L758-L814